src/sql/src/names.rs

```diff
@@ -520,11 +533,12 @@ impl<'a> Fold<Raw, Aug> for NameResolver<'a> {
                     Ok(id) => id,
                     Err(e) => {
                         self.status = Err(e.into());
-                        GlobalId::User(0)
+                        return error_response;
                     }
                 };
-                if self.status.is_ok() && self.catalog.try_get_item_by_id(&gid).is_none() {
+                if self.catalog.try_get_item_by_id(&gid).is_none() {
```

Comment:
This isn't a huge deal, but I believe the predicate you removed likely served the purpose of making sure this kind of vague error didn't overwrite other, more specific errors--or more generally was trying to avoid later errors' propagating from clobbering earlier ones. Did you encounter something specific that led you to see this as the wrong approach?

Reply:
I had found it a bit confusing that we were letting errors propagate and implicitly returning garbage values. For example above if we fail to parse the `gid` we just set it to `0` and carry on, but any caller of this function should see that `self.status` is an error and ignore the additional work we do.

Instead I thought it would be more obvious to always immediately return on an error. Which means that an error can't propagate down here. Unless there's some recursive calls in here that I missed.

Reply:
Though it's possible that my take is wrong, in which case I can revert it back to letting errors propagate.

Reply:
Ah, my ask then would be to propagate this early return to all of the sites where we assign the error status. Don't have a preference whether or not we early return, but the mechanism should be consistent––ideally everywhere, but at least on each struct.

Reply:
Ok, I finally see what the issue is, I'm just going to add the condition back. I wasn't thinking about how a single `NameResolver` may call this function many times, and errors from previous invocations can propagate to later invocations.

In fact I think probably any time `self.status` is changed, it should be surrounded by a check to `is_ok()`, which isn't currently the case.

Reply:
I added the check back and also added it to a couple of other places in this method.